### PR TITLE
EMANorm adjusting learning rate

### PR DESCRIPTION
**Pull request by levmckinney**
## Description

Addresses #545

## Testing

There is already testing for EMANorm though it seems like it will need to be updated to address the new update rule.

### CODE REVIEW

File: tests/util/test_networks.py [hunk 1]
```diff
@@ -18,8 +18,8 @@
 ]
 
 
-class EMANormIncremental(networks.EMANorm):
-    """EMA Norm with every new instance added incrementally in a for loop."""
+class EMANormAlgorithm2(networks.EMANorm):
+    """EMA Norm using algorithm 2."""
```
Comment (AdamGleave) [inline tests/util/test_networks.py:22]: Cite the PDF doc? Algorithm 2 is meaningless as a reference without it.

**Review by AdamGleave — APPROVED** [review]
LGTM apart from one small suggestion. This adjustment is quite elegant. I did not verify that the derivation of Algorithm 3 still holds when the weights are variable but the proofs looked reasonable.

@taufeeque9 you've written a lot of the surrounding code, do you mind taking a quick look at this stage?

**Review by taufeeque9 — APPROVED** [review]
LGTM. 
One minor concern: we initialize `running_mean` and `running_var` as `torch.empty` tensors, which means they have uninitialized data. On the first iteration, we add to these tensors the delta tensors which contain the negative of the respective tensors. So logically, it wouldn't create any problems. But would it be cleaner to initialize the tensors with zeros instead?

**Comment by levmckinney** [comment]
> LGTM. One minor concern: we initialize `running_mean` and `running_var` as `torch.empty` tensors, which means they have uninitialized data. On the first iteration, we add to these tensors the delta tensors which contain the negative of the respective tensors. So logically, it wouldn't create any problems. But would it be cleaner to initialize the tensors with zeros instead?

Good point, don't want to be messing with utilized data. I'll initialize them to be zeros.

**Pull request merged**